engine/src/main/java/org/terasology/rendering/opengl/GLSLMaterial.java

```diff
@@ -152,6 +156,43 @@ public void recompile() {
         if (shaderParameters != null) {
             shaderParameters.initialParameters(this);
         }
+
+        //solution due to problems with uniforms.
```

Comment:
It would be great if you could be a little more specific in this comment here. Maybe refer to issue #966 and/or wrap it in a method `rebindMaterial()`?
People will then know why this code part exists and what happens if they consider removing it.